src/clinical/service-worker-thread/tasks.ts

```diff
@@ -278,11 +279,22 @@ function extractEntityDataFromDonors(
       if (completionRecord && completionRecord.coreCompletion?.specimens > 0) {
         const donorSpecimenData = specimens.filter(dnaSampleFilter);
 
+        const specimenStats = calculateSpecimenCompletionStats(donorSpecimenData);
+
+        if (specimenStats.normalSpecimens < 1) {
+          const normalRegistrations = filterTumourNormalRecords(donorSpecimenData, 'Normal');
+          specimenStats.normalSpecimens = -normalRegistrations.length;
+        }
+        if (specimenStats.tumourSpecimens < 1) {
+          const tumourRegistrations = filterTumourNormalRecords(donorSpecimenData, 'Tumour');
+          specimenStats.tumourSpecimens = -tumourRegistrations.length;
+        }
+
```

Comment:
I think using this one property to convey all these interpretations of the core completion data will lead to confusion. As we discussed, it would be better to separate all our data in the response to clearly state hte number of registrations, submissions, and the completion score separately.

Reply:
Updated w/ a Specimen Core Completion type + an additional entityData field extendable for future scenarios